hudi-common/src/main/java/org/apache/hudi/common/table/read/buffer/FileGroupRecordBufferInitializer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.read.buffer;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.engine.HoodieReaderContext;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.read.BaseFileUpdateCallback;
+import org.apache.hudi.common.table.read.HoodieReadStats;
+import org.apache.hudi.common.table.read.InputSplit;
+import org.apache.hudi.common.table.read.ReaderParameters;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.storage.HoodieStorage;
+
+import java.util.List;
+
+/**
+ * This interface defines the contract for initializing a {@link FileGroupRecordBuffer} for a given file group.
+ * The implementation is expected to establish the record buffer and populate it with the records from the log files.
+ * @param <T> the engine specific record type
+ */
+public interface FileGroupRecordBufferInitializer<T> {
```

Comment:
maybe `Loader` instead of `Initializer` since it loads the records from the files into the "buffer" for use?  `Initializer` makes it feel like a factory for initializing an instance.

Reply:
Ok, updating to loader